backend/src/dbutil.ts

```diff
@@ -1,18 +1,126 @@
-import * as dt from "../schema/heat_pump_water_heater";
+import * as dt from "../schema/appliance";
 
-const dummyWaterHeater: dt.HeatPumpWaterHeater = {
+export function findWaterHeater(tankCapacity: number, uniformEnergyFactor: number, firstHourRating: number) {
+  return dummyWaterHeaters;
+}
+
+export function findDryer(soundLevel: number, combinedEnergyFactor: number, capacity: number) {
+  return dummyDryers;
+}
+
+const dummyWaterHeaters: dt.HeatPumpWaterHeater[] = [{
+  brandName: "Rheem",
+  modelNumber: "GEH50DFEJ2RA",
+  modelVairant: "701460",
```

Comment:
Now I'm a bit confused. These have the modelVariant field, but no modelType field. But we are still asking the frontend to specify modelType.

Isn't it simplest if there is one large set of fake data, and the frontend specifies modelType, and then you simply filter to the right models based on the value of each appliance's modelType?

E.g. 
```
const selectedModels = allModels.filter((model) => model.modelType === req.query['modelType']
```

In any case, if the field modelVariant is restricted by an enum in the schema, then a value of "701460" is not a valid modelVariant.

Reply:
I see where the confusion comes from.

When I first merged the change for the dummy API, this field was incorrectly named as `modelType` but instead it should be `modelVariant` as per the spec (https://github.com/datu925/electric-machines/blob/main/data/heat-pump-water-heaters/rheem/proterra-plugin-120v-shared/raw.pdf)
![image](https://github.com/datu925/electric-machines/assets/24417119/3d77d4c7-4fa3-43db-90ac-e3d207a046f3)

And we required the front-end to specify the modelType (I think it should also be renamed as `applianceType` for clearer interpretation) to filter based on the data requested, I am planning to add json validation and serialisation as the searching criteria differs based on different appliance types.
https://github.com/datu925/electric-machines/blob/eef814db04c8c497027abae7ccd06d1d055aa0ac/backend/src/server.ts#L22-L26

On the storage end, I think the best approach is to have a different datatable for each of the appliance type as we already have different schema for each of the appliance.

Reply:
Got it. In that case, there is still a problem in that the modelVariant in the JSON schema is constrained by an enum which is no longer appropriate (because it refers to the model types). I'll fix this in a quick follow-up. There's also the typo I mentioned earlier.

I agree with you on the change from modelType to applianceType.

In the interest of moving things forward, I'm going to merge this and then fix those things immediately after.